code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm

```diff
@@ -6,113 +6,179 @@
 /datum/action/xeno_action/activable/transfer_plasma/drone
 	plasma_transfer_amount = PLASMA_TRANSFER_AMOUNT * 2
 
+/datum/action/xeno_action/create_jelly/slow
+	cooldown_timer = 45 SECONDS
+
 // ***************************************
-// *********** Acidic salve
+// *********** Essence Link
+// ***************************************
+/datum/action/xeno_action/activable/essence_link
+	name = "Essence Link"
+	action_icon_state = "healing_infusion"
+	mechanics_text = "Link to a xenomorph. This changes some of your abilities, and grants them and you both various bonuses."
+	cooldown_timer = 1
+	plasma_cost = 0
+	target_flags = XABB_MOB_TARGET
+	keybind_signal = COMSIG_XENOABILITY_ESSENCE_LINK
+	alternate_keybind_signal = COMSIG_XENOABILITY_ESSENCE_LINK_REMOVE
+	/// Time it takes for the link to form.
+	var/link_delay = 3 SECONDS
+	/// Used to determine whether there is an existing Essence Link or not. Also allows access to its vars.
+	var/datum/status_effect/stacking/essence_link/existing_link
+	/// If there is an existing link, this var will contain the link's target.
+	var/mob/living/carbon/xenomorph/linked_target
+	/// Used to determine whether Enhancement is already active or not. Also allows access to its vars.
+	var/datum/status_effect/drone_enhancement/existing_enhancement
+
+/datum/action/xeno_action/activable/essence_link/can_use_ability(mob/living/carbon/xenomorph/target, silent = FALSE, override_flags)
+	var/mob/living/carbon/xenomorph/X = owner
+	existing_link = X.has_status_effect(STATUS_EFFECT_XENO_ESSENCE_LINK)
+	if(existing_link)
+		linked_target = existing_link.link_target
+		existing_enhancement = linked_target.has_status_effect(STATUS_EFFECT_XENO_ENHANCEMENT)
```

Comment:
This is here to prevent runtimes. The ability would runtime sometimes if there was no `existing_link`, since it was trying to get `link_target` out of `0`.

Reply:
Should be fixed in next commit.